pkg/odo/cli/dev/dev.go

```diff
@@ -0,0 +1,131 @@
+package dev
+
+import (
+	"fmt"
+	dfutil "github.com/devfile/library/pkg/util"
+	"github.com/redhat-developer/odo/pkg/devfile/adapters/kubernetes"
+	"github.com/redhat-developer/odo/pkg/envinfo"
+	"github.com/redhat-developer/odo/pkg/odo/cli/component"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"os"
+	"path/filepath"
+)
+
+// RecommendedCommandName is the recommended command name
+const RecommendedCommandName = "dev"
+
+type DevOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Variables
+	ignorePaths []string
+}
+
+func NewDevOptions() *DevOptions {
+	return &DevOptions{}
+}
+
+var devExample = templates.Examples(`
+	# Deploy component to the development cluster
+	%[1]s
+`)
+
+func (o *DevOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *DevOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
+	var err error
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline).NeedDevfile(""))
+	if err != nil {
+		return fmt.Errorf("unable to create context: %v", err)
+	}
+
+	devfileExists := util.CheckPathExists(o.Context.GetDevfilePath())
+	if !devfileExists {
+		return fmt.Errorf("the current directory doesn't contain a devfile")
+	}
+
+	envFileInfo, err := envinfo.NewEnvSpecificInfo("")
+	if err != nil {
+		return fmt.Errorf("unable to retrieve configuration information: %v", err)
+	}
+	if !envFileInfo.Exists() {
+		var cmpName string
+		cmpName, err = component.GatherName(o.EnvSpecificInfo.GetDevfileObj(), o.GetDevfilePath())
+		if err != nil {
+			return fmt.Errorf("unable to retrieve component name: %w", err)
+		}
+		err = envFileInfo.SetComponentSettings(envinfo.ComponentSettings{Name: cmpName, Project: o.GetProject(), AppName: "app"})
+		if err != nil {
+			return fmt.Errorf("failed to write new env.yaml file: %w", err)
+		}
+	} else if envFileInfo.GetComponentSettings().Project != o.GetProject() {
+		err = envFileInfo.SetConfiguration("project", o.GetProject())
+		if err != nil {
+			return fmt.Errorf("failed to update project in env.yaml file: %w", err)
+		}
+	}
```

Comment:
This part is quite obscure to me. What is the final purpose?

Reply:
Added comments. Does that look good now?